### ISSUE TITLE
[Enhancement]: Support changing the theme of electron-apps on konsole or os-level 

### ISSUE THREAD

**Issue by slynobody** [electron/electron#37636]
### Preflight Checklist

- [X] I have read the [Contributing Guidelines](https://github.com/electron/electron/blob/main/CONTRIBUTING.md) for this project.
- [X] I agree to follow the [Code of Conduct](https://github.com/electron/electron/blob/main/CODE_OF_CONDUCT.md) that this project adheres to.
- [X] I have searched the [issue tracker](https://www.github.com/electron/electron/issues) for a bug report that matches the one I want to file, without success.

### Electron Version

21.4.3

### What operating system are you using?

Other Linux

### Operating System Version

Debian Sid

### What arch are you using?

x64

### Last Known Working Electron version

21.4.3

### Expected Behavior

change the theme of an electron-apps through linux konsole or os-level on the fly (f.e. through slider) *not* inside the app, f.e. triggered by os-defaults (currently not honoring them) or a through konsole as well as through external light-/dark-automation-tool for the whole system (de, gtk, qt, etc., electron seems to be the last not working / respecting os-defaults).

as workaround could you please give some working konsole command for changing electron-theme (from outside, not through electron-menus) on linux-systems (at least to adapt it to os-defaults)?

### Actual Behavior

you simply cant change the theme of electron-apps on konsole or os-level currently (electron is the last platform on linux with this problem) which renders all external tools on this useless.

new linux-users trust their os-options on light-/dark-change and see their favorite apps not changing (and thinking linux is ...)

### Additional Information

also see #36040, #33635, #28887 (Ubuntu, Fedora and others, seems to affect *all* linux distros)
also see issues f.e. on electron-apps like signal: https://github.com/signalapp/Signal-Desktop/issues/5091

**Comment by slynobody** [comment]
Is this really an 'enhancement'? electron simply respects every other os' defaults on this - beside linux under wayland (x11 works, but x11 gets ancient). There seem toalso some bug-related sides on this codewise?

**Comment by charlesneimog** [comment]
Any update on this?

**Comment by surajsharma** [comment]
> Is this really an 'enhancement'? electron simply respects every other os' defaults on this - beside linux under wayland (x11 works, but x11 gets ancient). There seem toalso some bug-related sides on this codewise?

yes, this is an enhancement. the electron app should respect OS theme during runtime

**Comment by bellackn** [comment]
I think this _might_ be the right place to slide in with my issue/request. I have opened this with the XFCE guys, but they told me it's not an XFCE issue, and the only common denominator of the affected apps is Electron I believe. So...

I'm using some Electron apps on Arch Linux/Manjaro with XFCE DE. The dark mode/system theme doesn't get picked up correctly when using xfconf-query, **unless the Appearance window is open**.

_Disclaimer: I'm not too deep into Electron/XFCE/DEs in general, so please bear with me if I use wrong vocabulary or anything._

This is a weird one. I have this script that I use to change my theme and background image from light to dark.

<details>
<summary>theme-toggle.sh (click to expand)</summary>

```
#!/usr/bin/env bash

# light
gtk_light="Adwaita"
xfwm_light="Adwaita"
icons_light="Papirus-Light"
desktop_bg_light="/usr/share/backgrounds/xfce/mountains-1412683.jpg"
alacritty_theme_light="papercolor_light"

# dark
gtk_dark="Adwaita-dark"
xfwm_dark="Adwaita-dark"
icons_dark="Papirus-Dark"
desktop_bg_dark="/usr/share/backgrounds/manjaro-wallpapers-18.0/manjaro-cat.jpg"
alacritty_theme_dark="challenger_deep"

# current mode determined by gtk theme
gtk="$(xfconf-query -c xsettings -p /Net/ThemeName)"

if [[ "$gtk" = "$gtk_light" ]]; then
  xfconf-query -c xsettings -p /Net/ThemeName -s "$gtk_dark"
  xfconf-query -c xfwm4 -p /general/theme -s "$xfwm_dark"
  xfconf-query -c xsettings -p /Net/IconThemeName -s "$icons_dark"
  xfconf-query -c xfce4-panel -p /panels/dark-mode -s true
  xfconf-query -c xfce4-desktop -p /backdrop/screen0/monitoreDP/workspace0/last-image -s "$desktop_bg_dark"
  xfconf-query -c xfce4-desktop -p /backdrop/screen0/monitorDisplayPort-1/workspace0/last-image -s "$desktop_bg_dark"
  sed -i -e "s/$alacritty_theme_light/$alacritty_theme_dark/" ~/.config/alacritty/alacritty.toml
else
  xfconf-query -c xsettings -p /Net/ThemeName -s "$gtk_light"
  xfconf-query -c xfwm4 -p /general/theme -s "$xfwm_light"
  xfconf-query -c xsettings -p /Net/IconThemeName -s "$icons_light"
  xfconf-query -c xfce4-panel -p /panels/dark-mode -s false
  xfconf-query -c xfce4-desktop -p /backdrop/screen0/monitoreDP/workspace0/last-image -s "$desktop_bg_light"
  xfconf-query -c xfce4-desktop -p /backdrop/screen0/monitorDisplayPort-1/workspace0/last-image -s "$desktop_bg_light"
  sed -i -e "s/$alacritty_theme_dark/$alacritty_theme_light/" ~/.config/alacritty/alacritty.toml
fi

exit 0
```

</details>

This works for the XFCE panel and windows title bars etc. - but some apps (like VS Codium, the Signal desktop app, the Vivaldi browser, ...) are not able to pick up the system theme after running this script. This means that if, say, VS Codium is in light mode (and set up in a way that it picks up the system theme to toggle between IDE themes), it will _stay_ in light mode, no matter how often I run this script.

**There is an exception though:** When I change the theme via the desktop GUI (Settings Manager > Apperance > Style), everything works like normal and all apps pick up the change correctly.

So I came up with this workaround in my script that makes it work as expected:

```
[...]
# workaround for apps picking up the correct color scheme...
xfce4-appearance-settings &
pid=$!
sleep 0.5
[...]
kill $pid
```

This opens the Appearance settings window, then runs the commands mentioned further above. When done this way, all apps pick up the new system theme.

I have tried with `sleep` amounts lower than 0.5, but then it also didn't work reliably. It's a pretty non-deterministic issue.
